### PR TITLE
feat: migrated to the latest SDK and thus supported export to OTLP

### LINKED ISSUE #92: Add the capability to use the OTLP protocol for sending metrics and traces.
Add the capability to use the OTLP protocol for sending metrics and traces. I suggest use [environment varible](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#exporter-selection) to configure exporter parameters. This involves enabling the system to send metrics and traces to the desired endpoints defined via environment variables OTEL_EXPORTER_OTLP_METRICS_ENDPOINT and OTEL_EXPORTER_OTLP_TRACES_ENDPOINT. Additionally, include TraceId and SpanId in logs when tracing is enabled.

### PR DESCRIPTION
**Pull request by adubovik**
Resolves https://github.com/epam/ai-dial-adapter-openai/issues/92

### CODE REVIEW

**Comment by adubovik** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/466578)
>Environment URL: https://chat-ai-dial-adapter-openai-pr-99.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-openai-pr-99.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-openai-pr-99.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-openai-pr-99.nightly-test.deltixhub.io
>E2E tests status: success

File: pyproject.toml [hunk 1]
```diff
@@ -30,13 +30,13 @@ pydantic = "^1.10.12"
 numpy = "^1.26.0" # need for openai embeddings
 pillow = "^10.3.0"
 azure-identity = "^1.15.0"
-aidial-sdk = "^0.7.0"
+aidial-sdk = {version = "0.8.0", extras = ["telemetry"]}
```
Comment (vladisavvv) [inline pyproject.toml]: let's change to ^0.8.0 to get our security patches

Comment (adubovik) [reply pyproject.toml]: Fixed

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-openai-pr-99.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by vladisavvv — APPROVED** [review]

**Pull request merged**